packages/wrangler/src/api/startDevWorker/BaseController.ts

```diff
@@ -68,7 +69,15 @@ export type ControllerEventMap = {
 export abstract class Controller<
 	EventMap extends ControllerEventMap = ControllerEventMap,
 > extends TypedEventEmitterImpl<EventMap> {
+	#tearingDown = false;
+	async teardown(): Promise<void> {
+		this.#tearingDown = true;
+	}
 	emitErrorEvent(data: ErrorEvent) {
+		if (this.#tearingDown) {
+			logger.debug("Suppressing error event during teardown");
+			return;
+		}
```

Comment:
I cannot prove it, since it is very non-deterministic but I think this helps a lot with reduction of flakes.

There appear to be test scenarios where the file watching handlers are not torn down "quickly enough" and then cause errors at the end of test runs, since the underlying directories have been removed.

This results in "unhandled promise rejection" errors in the tests.

There is no benefit in user's knowing about errors during teardown. In real-life teardown happens as the process is ending so the user is expecting termination anyway.

Reply:
Since all the controllers have this potential problem, not just ConfigController, this logic is now shared between them all.

I am pretty sure that the BundleController tests occasionally exhibited this problem in the past, for instance.

Reply:
Done

Reply:
[436e254](https://github.com/cloudflare/workers-sdk/pull/10862/commits/436e254d8ef876c661a636be9367c23e3174f3d6)